test/i18n_test.rb

```diff
@@ -287,6 +287,34 @@ def setup
     assert_equal I18n.config.available_locales.size * 2, I18n.config.available_locales_set.size
   end
 
+  test "interpolation_keys returns an array of keys" do
+    store_translations(:en, "example_two" => "Two interpolations %{foo} %{bar}")
+    assert_equal ["foo", "bar"], I18n.interpolation_keys("example_two")
+  end
+
+  test "interpolation_keys returns an empty array when no interpolations " do
+    store_translations(:en, "example_zero" => "Zero interpolations")
+    assert_equal [], I18n.interpolation_keys("example_zero")
+  end
+
+  test "interpolation_keys returns an empty array when missing translation " do
+    assert_equal [], I18n.interpolation_keys("does-not-exist")
+  end
+
+  test "interpolation_keys returns an empty array when nested translation" do
+    store_translations(:en, "example_nested" => { "one" => "One %{foo}", "two" => "Two %{bar}" })
+    assert_equal [], I18n.interpolation_keys("example_nested")
```

Comment:
In this scenario, `I18n.t("example_nested")` _does not require/use any interpolation keys_ (even though the child translations do). Therefore I feel it's correct for the method to return `[]`.